mackerel-plugin-windows-server-sessions/lib/sessions.go

```diff
@@ -1,39 +1,50 @@
+//go:build windows
+
 package mpwindowsserversessions
 
 import (
-	"encoding/csv"
 	"flag"
-	"os/exec"
-	"strconv"
+	"os"
 	"strings"
 
+	"github.com/yusufpapurcu/wmi"
+
 	mp "github.com/mackerelio/go-mackerel-plugin-helper"
 )
 
 // WindowsServerSessionsPlugin store the name of servers
 type WindowsServerSessionsPlugin struct {
 }
 
-func getCounts() (map[string]int, error) {
-	// WMIC PATH Win32_PerfFormattedData_PerfNet_Server GET ServerSessions /FORMAT:CSV
-	output, err := exec.Command("WMIC", "PATH", "Win32_PerfFormattedData_PerfNet_Server", "GET", "ServerSessions", "/FORMAT:CSV").Output()
-	if err != nil {
+// cf.) https://learn.microsoft.com/en-us/previous-versions/aa394265(v=vs.85)
+type Win32_PerfFormattedData_PerfNet_Server struct {
+	ServerSessions uint32
+}
+
+type ServerSessionsCount struct {
+	Node           string
+	ServerSessions uint32
+}
+
+func getCounts() ([]ServerSessionsCount, error) {
+	var dst []Win32_PerfFormattedData_PerfNet_Server
+	q := wmi.CreateQuery(&dst, "")
+	if err := wmi.Query(q, &dst); err != nil {
 		return nil, err
 	}
-	r := csv.NewReader(strings.NewReader(string(output[1:])))
-	records, err := r.ReadAll()
+
+	if len(dst) == 0 {
+		return []ServerSessionsCount{}, nil
+	}
+
+	node, err := os.Hostname()
 	if err != nil {
 		return nil, err
 	}
-	counts := make(map[string]int)
-	for _, record := range records[1:] {
-		name := strings.TrimSpace(record[0])
-		n, err := strconv.Atoi(strings.TrimSpace(record[1]))
-		if err != nil {
-			continue
-		}
-		counts[name] = n
+	counts := []ServerSessionsCount{
+		{Node: node, ServerSessions: dst[0].ServerSessions},
```

Comment:
The plugin was originally implemented as if it could retrieve metrics for multiple nodes, but in reality it can only retrieve metrics for one node at most.